### PR TITLE
feat(events-subscription): allow to instantly refresh customizations when they change

### PR DESCRIPTION
**Pull request by Thenkei**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/agent/src/agent.ts [hunk 1]
```diff
@@ -88,6 +88,11 @@ export default class Agent<S extends TSchema = TSchema> extends FrameworkMounter
         : Promise.resolve(),
     ]);
 
+    if (!restart) {
+      this.options.forestAdminClient.subscribeToServerEvents();
+      this.options.forestAdminClient.onRefreshCustomizations(this.start.bind(this, null, true));
```
Comment (Thenkei) [inline packages/agent/src/agent.ts:93]: On RefreshCustomizations we launch a start but omit to `subscribeToServerEvents` and to re-listen to `RefreshCustomizations`

File: packages/agent/src/agent.ts [hunk 2]
```diff
@@ -70,7 +70,7 @@ export default class Agent<S extends TSchema = TSchema> extends FrameworkMounter
   /**
    * Start the agent.
    */
-  override async start(): Promise<void> {
+  override async start(_, restart = false): Promise<void> {
```
[the remark this answers is not in the document]
Comment (ghusse) [reply packages/agent/src/agent.ts:73]: Maybe it can be another method here, instead of passing arguments?

File: packages/agent/src/types.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ export type AgentOptions = {
   schemaPath?: string;
   typingsPath?: string | null;
   typingsMaxDepth?: number;
+  useServerEvents?: boolean;
```
Comment (ghusse) [inline packages/agent/src/types.ts:18]: If we want to provide an option here (to discuss with the DevXP), I think the name should be more explicit about what it does and not how it works.

File: packages/agent/src/agent.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Thenkei) [reply packages/agent/src/agent.ts:73]: It's only a POC for now. 

But it wasn't really easy to do without duplicating code or refactoring more things.

**Pull request closed without merging**